openhantek/src/main.cpp

```diff
@@ -84,10 +84,10 @@ int main(int argc, char *argv[]) {
         useGles = p.isSet(useGlesOption);
     }
 
-    GlScope::fixOpenGLversion(useGles ? QSurfaceFormat::OpenGLES : QSurfaceFormat::OpenGL);
-
     QApplication openHantekApplication(argc, argv);
 
+    GlScope::fixOpenGLversion(useGles ? QSurfaceFormat::OpenGLES : QSurfaceFormat::OpenGL);
```

Comment:
fixOpenGLversion must be called before the (real) QApplication is initalized. Because the `QSurfaceFormat` can only be set before the Qt application is initialized. That's what I have read at least.

Reply:
Got it, thanks. This requirement invalidates recent approach to auto-detection, I'd better revoke entire PR and re-start from scratch.